setup.sh

```diff
@@ -110,6 +110,100 @@ function check_dependencies()
   fi
 }
 
+# This function checks and installs kernel build dependencies.
+# It supports three Linux distributions: Debian, Arch Linux, and Fedora.
+#
+# Returns:
+# return 0: All necessary dependencies are already installed or installation is successful.
+# return 1 (EPERM): Unsupported OS type.
+function check_and_install_kernel_build_dependencies()
+{
+  local package_list
+  local cmd
+  local distro
+  local kernel_build_deps_file
+  local installed
+
+  distro=$(detect_distro '/')
+
+  kernel_build_deps_file="${DOCUMENTATION}/dependencies/kernel_build/${distro}.dependencies"
+
+  case "$distro" in
+    arch*)
+      while IFS='' read -r package; do
+        installed=$(pacman -Ql "$package" &> /dev/null)
+        [[ "$?" != 0 ]] && package_list="${package} ${package_list}"
+      done < "$kernel_build_deps_file"
+      cmd="pacman -Sy --noconfirm ${package_list}"
+      ;;
+    debian*)
+      while IFS='' read -r package; do
+        installed=$(dpkg-query -W --showformat='${Status}\n' "$package" 2> /dev/null | grep -c 'ok installed')
+        [[ "$installed" -eq 0 ]] && package_list="${package} ${package_list}"
+      done < "$kernel_build_deps_file"
+      cmd="apt install -y ${package_list}"
+      ;;
+    fedora*)
+      while IFS='' read -r package; do
+        installed=$(rpm -q "$package" &> /dev/null)
+        [[ "$?" -ne 0 ]] && package_list="${package} ${package_list}"
+      done < "$kernel_build_deps_file"
+      cmd="dnf install -y ${package_list}"
+      ;;
+    *)
+      complain "Unsupported OS type: ${distro}"
+      return 1 # EPERM
+      ;;
+  esac
+
+  if [[ -z "$package_list" ]]; then
+    warning 'All necessary kernel build dependencies are already installed.'
+    return 0
+  fi
+
+  if [[ -n "$package_list" ]]; then
+    if [[ "$FORCE" == 0 ]]; then
+      if [[ $(ask_yN "The following packages are required to build the kernel: ${package_list}"$'\nMay we install them?') =~ '0' ]]; then
+        complain 'Aborting installation. Kernel build dependencies not installed.'
+        exit 125 # ECANCELED
+      fi
+    fi
+
+    # Install system packages
+    if [[ "$EUID" -eq 0 ]]; then
+      eval "$cmd"
+    else
+      eval "sudo ${cmd}"
+    fi
+
+    # Installation failed...
+    if [[ "$?" -ne 0 ]]; then
+      complain '[ERROR] Dependencies installation to build the kernel has failed. Aborting installation...'
+      exit "$?"
+    fi
+  fi
+}
+
+# This function initiates the process to ensure kernel build dependencies are
+# installed.
+function install_kernel_dev_deps()
+{
+  local distro
+  local ret
+
+  distro=$(detect_distro '/')
+
+  # Check if distro is equal to 'none'
+  if [[ "$distro" == 'none' ]]; then
+    complain 'Support for this distro is not available yet.'
+    exit 22 # EINVAL
+  fi
+
+  # Use check_and_install_kernel_build_dependencies() to handle dependency
+  # installation
+  check_and_install_kernel_build_dependencies
```

Comment:
As you are detecting the distro above and using the same expression to obtain it again inside `check_and_install_kernel_build_dependencies`, can't we pass `distro` as an argument here?